server/src/main/java/io/crate/execution/ddl/tables/RenameColumnTask.java

```diff
@@ -0,0 +1,276 @@
+                partitionedBy.addFirst(oldName.replaceFirst(renameColumn.refToRename().column().fqn(), renameColumn.newName().fqn()));
+            }
+        }
+    }
+
+    private static void renameKeysFromPrimaryKeys(Map<String, Object> mapping,
+                                                  AnalyzedAlterTableRenameColumn.RenameColumn renameColumn) {
+        List<String> primaryKeys = (List<String>) mapping.remove("primary_keys");
+        List<String> renamed = new ArrayList<>(primaryKeys.size());
+        for (String primaryKey : primaryKeys) {
+            // TODO: Use ColumnRenameVisitor
+            if (primaryKey.equals(renameColumn.refToRename().column().fqn())) {
+                renamed.add(renameColumn.newName().fqn());
+            } else if (primaryKey.startsWith(renameColumn.refToRename().column().fqn())) {
+                renamed.add(primaryKey.replaceFirst(renameColumn.refToRename().column().fqn(), renameColumn.newName().fqn()));
+            } else {
+                renamed.add(primaryKey);
+            }
+        }
+        mapping.put("primary_keys", renamed);
+    }
+
+    private static void renameKeysFromCheckConstraints(Map<String, Object> mapping,
+                                                       AnalyzedAlterTableRenameColumn.RenameColumn renameColumn) {
+        var checks = (Map<String, String>) mapping.get("check_constraints");
+        for (var e : checks.entrySet()) {
+            var checkExpr = SqlParser.createExpression(e.getValue());
+            var renamedCheckExpr = new ColumnRenameVisitor().rename(
+                checkExpr,
+                Map.entry(SqlParser.createExpression(renameColumn.refToRename().column().sqlFqn()),
+                    SqlParser.createExpression(renameColumn.newName().sqlFqn())));
+            // TODO: make sure any quotings are preserved.
+            checks.put(e.getKey(), renamedCheckExpr.toString());
+        }
+    }
+
+    private static void renameKeysFromGeneratedExpressions(Map<String, Object> mapping,
+                                                           AnalyzedAlterTableRenameColumn.RenameColumn renameColumn) {
+        var generatedColumns = (Map<String, String>) mapping.get("generated_columns");
+        for (var e : generatedColumns.entrySet()) {
+            var genExpr = SqlParser.createExpression(e.getValue());
+            var renamedGenExpr = new ColumnRenameVisitor().rename(
+                genExpr,
+                Map.entry(SqlParser.createExpression(renameColumn.refToRename().column().sqlFqn()),
+                    SqlParser.createExpression(renameColumn.newName().sqlFqn())));
+            // TODO: make sure any quotings are preserved.
+            generatedColumns.put(e.getKey(), renamedGenExpr.toString());
+        }
+    }
+
+    /* Renames any columns within the expression. */
+    private static class ColumnRenameVisitor extends AstVisitor<Expression, Map.Entry<Expression, Expression>> {
```

Comment:
This is needed to re-parse checks/generate expressions like - 
`abs(a + o['a']))`, `(o['a'] > a)` to safely rename columns in between. Any suggestions on this? If this seems ok, I will refine this further to more generic `ExpressionRewriter` and let `ColumnRenameVisitor` to extend it.


The only trouble right now is with quoting. `Expression.toString()` adds extra quotes. What comes to my mind in order to resolve quoting is to create another visitor that converts `Expression` to `ColumnIdent`(then to `String`) but I will see if I can avoid this.

Reply:
Thank you, this has been resolved. Please check the updates since the last review.